plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/contributors/FirSuperMemberCompletionContributor.kt

```diff
@@ -0,0 +1,94 @@
+    basicContext: FirBasicCompletionContext,
+    priority: Int
+) : FirCompletionContributorBase<FirNameReferencePositionContext>(basicContext, priority) {
+    override fun KtAnalysisSession.complete(positionContext: FirNameReferencePositionContext) = with(positionContext) {
+        val superReceiver = explicitReceiver as? KtSuperExpression ?: return
+        val expectedType = nameExpression.getExpectedType()
+        val visibilityChecker = CompletionVisibilityChecker.create(basicContext, positionContext)
+        val possibleReceiverScope = superReceiver.getKtType()?.getTypeScope() ?: return
+        val nonExtensionMembers = collectNonExtensions(possibleReceiverScope, visibilityChecker, scopeNameFilter)
+        collectDelegateCallToSuperMember(superReceiver, nonExtensionMembers, expectedType)
+    }
+
+    private fun KtAnalysisSession.collectDelegateCallToSuperMember(
+        superReceiver: KtSuperExpression,
+        nonExtensionMembers: Sequence<KtCallableSymbol>,
+        expectedType: KtType?
+    ) {
+        // A map that contains all containing functions as values, each of which is indexed by symbols it overrides. For example, consider
+        // the following code
+        // ```
+        // class A : Runnable {
+        //   override fun run() {
+        //     val o = object: Callable<String> {
+        //       override fun call(): String {
+        //         super.<caret>
+        //       }
+        //     }
+        //   }
+        // }
+        // ```
+        // The map would contain
+        //
+        // * Runnable.run -> A.run
+        // * Callable.call -> <anonymous object>.call
+        val superFunctionToContainingFunction = superReceiver
+            .parentsOfType<KtNamedFunction>(withSelf = false)
+            .flatMap { containingFunction ->
+                val original = basicContext.toFromOriginalFileMapper.toOriginalFile(containingFunction) ?: return@flatMap emptyList()
+                original
+                    .getFunctionLikeSymbol()
+                    .getAllOverriddenSymbols()
+                    .map { superFunctionSymbol ->
+                        superFunctionSymbol to original
+                    }
+            }.toMap()
+
+        if (superFunctionToContainingFunction.isEmpty()) return
+
+        for (callableSymbol in nonExtensionMembers) {
+            val matchedContainingFunction = superFunctionToContainingFunction[callableSymbol] ?: continue
+            if (callableSymbol !is KtFunctionSymbol) continue
+            if (callableSymbol.valueParameters.isEmpty()) continue
+            val args = matchedContainingFunction.valueParameters.mapNotNull {
+                val name = it.name ?: return@mapNotNull null
+                if (it.isVarArg) {
+                    "*$name"
+                } else {
+                    name
+                }
+            }
+            if (args.size < matchedContainingFunction.valueParameters.size) continue
```

Comment:
Why do we need this check?

Reply:
This is just to guard against the case that a `KtParameter` somehow doesn't have a name, in which case, we don't want to offer this completion.